cpp/libcudf_kafka/src/kafka_consumer.cpp

```diff
@@ -150,21 +150,29 @@ int64_t kafka_consumer::get_committed_offset(std::string const &topic, int parti
                "Failed retrieve Kafka committed offsets");
 
   int64_t offset = toppar_list[0]->offset();
-  return offset > 0 ? offset : 0;
+  return offset > 0 ? offset : -1001;
 }
 
-std::map<std::string, std::vector<int32_t>> kafka_consumer::list_topics() const
+std::map<std::string, std::vector<int32_t>> kafka_consumer::list_topics(
+  std::string specific_topic) const
 {
-  auto const md = [&]() {
+  auto const metadata = [&]() {
+    std::string errstr;
+    RdKafka::Topic const *const spec_topic =
+      specific_topic.empty()
+        ? nullptr
+        : RdKafka::Topic::create(consumer.get(), specific_topic, nullptr, errstr);
+
     RdKafka::Metadata *md;
-    CUDF_EXPECTS(RdKafka::ERR_NO_ERROR == consumer->metadata(true, nullptr, &md, default_timeout),
-                 "Failed to list_topics in Kafka broker");
+    CUDF_EXPECTS(
+      RdKafka::ERR_NO_ERROR ==
+        consumer->metadata(specific_topic.empty() ? true : false, spec_topic, &md, default_timeout),
```

Comment:
I think with this change you don't need to check the error string after the `create` call. Unless we want the `list_topics` call to fail if something goes wrong.
I recommend this change either way. IMO it's cleaner since we now don't repeat the `specific_topic.empty()` check.
```suggestion
        consumer->metadata(spec_topic == nullptr ? true : false, spec_topic, &md, default_timeout),
```

Reply:
I like this and think its the best way and prevents the extra check on the `errstr` after the call the create. I'll make the change